### ISSUE TITLE
How to get the different tests report in testng.xml

### ISSUE THREAD

**Issue by bharathramaswamy** [anshooarora/extentreports#276]
Hi Arora,

i am using the extent reports , where as i am running the same 8 tests in three broewsers and all the 27 tests are displayed contionusly .i want the tests to be specific about browsers .Please help me with the same .

below is the example 
<suite name="Parallel Test"  parallel="false" >

	<test name="Internet Explorer Test" >
	 <parameter name="browsername" value="internet explorer"/>
		<classes>
			 	<class name="class.SmokeTest"/>
		</classes>
	</test>
	
	<test name="Firefox Test" >
	 <parameter name="browsername" value="firefox"/>
		<classes>
			 	<class name="class.SmokeTest"/>
		</classes>
	</test>
	
	<test name="Chrome Test" >
	 <parameter name="browsername" value="chrome"/>
		<classes>
			 	<class name="class.SmokeTest"/>
		</classes>
	</test>
			</suite>

Please guide me how can i get the reports for the specific tests such as ie,firefox and chrome .

**Comment by anshooarora** [comment]
You can add a [category](http://extentreports.relevantcodes.com/java/version2/docs.html#assign-category) to the executing test by assigning the browser name.

**Comment by bharathramaswamy** [comment]
How can I do this for the XML I have given

**Comment by anshooarora** [comment]
You can read the parameter values in your method and assign.

**Comment by bharathramaswamy** [comment]
i am trying to do that , actually it displays all the tests appnded and i
am not able to find out in which browser its has failed

On Fri, Jan 15, 2016 at 10:06 PM, Anshoo Arora <notifications@github.com>
wrote:

> You can read the parameter values in your method and assign.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anshooarora/extentreports/issues/276#issuecomment-172009098>
> .
>



-- 
Bharath

**Comment by anshooarora** [comment]
This looks more like a TestNG question, and you should ask on SO or a relevant forum.

You can read values from your test framework and assign your tests using (as shown [here](http://testng.org/javadoc/org/testng/annotations/Parameters.html)):

```
@Parameters({ "xmlPath" })
@Test
public void verifyXmlFile(String path) { 
    test.assignCategory(xmlPath);
}
```
```
and in testng.xml:
<parameter name="xmlPath" value="account.xml" />
```

Categories can be filtered and they will provide you this information.

**Comment by bharathramaswamy** [comment]
Thank you so much man i was able to do with the different tests .

Thanks & Regards
Bharath

On Fri, Jan 15, 2016 at 11:01 PM, Anshoo Arora <notifications@github.com>
wrote:

> Closed #276 <https://github.com/anshooarora/extentreports/issues/276>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/anshooarora/extentreports/issues/276#event-516401943>.
>



-- 
Bharath
